crates/iceberg/src/error.rs

```diff
@@ -333,6 +335,28 @@ define_from_err!(
 
 define_from_err!(std::io::Error, ErrorKind::Unexpected, "IO Operation failed");
 
+/// Converts a timestamp in milliseconds to `DateTime<Utc>`, handling errors.
+///
+/// # Arguments
+///
+/// * `timestamp_ms` - The timestamp in milliseconds to convert.
+///
+/// # Returns
+///
+/// This function returns a `Result<DateTime<Utc>, Error>` which is `Ok` with the `DateTime<Utc>` if the conversion is successful,
+/// or an `Err` with an appropriate error if the timestamp is ambiguous or invalid.
+pub(crate) fn timestamp_ms_to_utc(timestamp_ms: i64) -> Result<DateTime<Utc>> {
+    match Utc.timestamp_millis_opt(timestamp_ms) {
+        chrono::LocalResult::Single(t) => Ok(t),
+        chrono::LocalResult::Ambiguous(_, _) => Err(Error::new(
+            ErrorKind::Unexpected,
```

Comment:
This this error be `InvalidData`? It's not quiet `Unexpected` to me.

Reply:
I think what @liurenjie1024 proposed above is most correct:
Ambiguous is unexpected - it would mean there is a fold in UTC time, which sounds unexpected to me.
None on the other hand is returned for different things. Among them is a gap in UTC time, which would be unexpected again, missing timezone data - also unexpected, but most likely an overflow - definitely invalid data.
So I am changing None -> DataInvalid
https://github.com/c-thiel/iceberg-rust/commit/abcb4cd0b664333e594706768ba63ed19ca1d7f6